tests/src/kakarot/instructions/test_system_operations.py

```diff
@@ -0,0 +1,229 @@
+import pytest
+from eth_account.account import Account
+from eth_utils import keccak
+from starkware.starknet.public.abi import get_selector_from_name
+
+from tests.utils.constants import CHAIN_ID
+from tests.utils.helpers import generate_random_private_key, pack_into_u64_words
+from tests.utils.syscall_handler import SyscallHandler
+from tests.utils.uint256 import int_to_uint256
+
+NONCE = 1
+MAGIC = 0x04
+CAIRO1_HELPERS_CLASS_HASH = 0xDEADBEEFABDE1
+
+
+@pytest.fixture
+def private_key():
+    return generate_random_private_key()
+
+
+@pytest.fixture
+def invoker_address(private_key):
+    return int(private_key.public_key.to_checksum_address(), 16)
+
+
+@pytest.fixture
+def message(invoker_address):
+    return {
+        "MAGIC": MAGIC.to_bytes(1, "big"),
+        "chainId": CHAIN_ID.to_bytes(32, "big"),
+        "nonce": NONCE.to_bytes(32, "big"),
+        "invokerAddress": invoker_address.to_bytes(32, "big"),
+        "commit": "commit".encode("utf-8").ljust(32, b"\x00"),
+    }
+
+
+def prepare_stack_and_memory(invoker_address, message, private_key):
+    serialized_msg = b"".join(message.values())
+    msg_hash = keccak(serialized_msg)
+    (messageHash, r, s, v, signature) = Account.signHash(msg_hash, private_key)
+    yParity = v - 27
+
+    addr_low, addr_high = int_to_uint256(invoker_address)
+    stack = [[addr_low, addr_high], [0, 0], [97, 0]]  # len, offset, invoker_address
+
+    offset = 0
+    memory = []
+    memory[offset : offset + 1] = [yParity]
+    memory[offset + 1 : offset + 33] = r.to_bytes(32, "big")
+    memory[offset + 33 : offset + 65] = s.to_bytes(32, "big")
+    memory[offset + 65 : offset + 97] = message["commit"]
+
+    return serialized_msg, msg_hash, stack, memory
+
+
+class TestSystemOperations:
+    class TestAuth:
+        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0, 1])
+        @SyscallHandler.patch("IAccount.get_nonce", lambda addr, data: [NONCE])
+        @SyscallHandler.patch("IAccount.bytecode", lambda addr, data: [0])
+        @SyscallHandler.patch(
+            "Kakarot_cairo1_helpers_class_hash",
+            CAIRO1_HELPERS_CLASS_HASH,
+        )
+        def test__exec_auth__should_pass_valid_signature(
+            self, cairo_run, private_key, invoker_address, message
+        ):
+            serialized_msg, msg_hash, stack, memory = prepare_stack_and_memory(
+                invoker_address, message, private_key
+            )
+
+            msg_hash_int = int.from_bytes(msg_hash, "big")
+
+            with SyscallHandler.patch(
+                "Kakarot_evm_to_starknet_address", invoker_address, 0x1234
+            ), SyscallHandler.patch(
```

Comment:
I think that we could add in the `SyscallHandler` class itself the fact that `library_call_keccak` is always define and works as expected, ie use the input `data` and returns what the helper returns

Reply:
Any guides on how you'd see that made?